File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def getDecimals(num):
    nums = 0;
    for i in range(1, int(num ** 0.5) + 1):
        if num % i == 0: nums += 2;
        if num / i == i: nums -= 1
    return nums

def solution(number, limit, power):
    # ver 2
    return sum([power if getDecimals(i + 1) > limit else getDecimals(i + 1) for i in range(number)])
    
    # ver 1
    # steel = 0;
    # for i in range(number):
    #     decimals = getDecimals(i + 1);
    #     if decimals > limit: steel += power
    #     else: steel += decimals
    # return steel
        

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function getDecimals(num) {
    let nums = 0;
    for(let i = 1; i <= Math.sqrt(num); i++){
        if(num % i === 0) nums += 2
        if(num / i === i) nums -= 1
    }
    return nums
}

function solution(number, limit, power) {
    return Array(number).fill(0).reduce((acc, _, idx) => {
        let decimals = getDecimals(idx + 1);
        return decimals > limit ? acc + power : acc + decimals
    }, 0)
}